### PR TITLE
added PHP 8.2

### PR DESCRIPTION
**Pull request by marcogiorgio**
# Usabilla PHP Docker Template

Reviewers: @usabilla/oss-docker

## Type

Please specify the type of changes being proposed:

| Q                 | A
| ----------------- | -----
| Documentation?    | no
| Dockerfile change?| no
| Build feature?    | no
| Apply CVE Patch?  | no
| Remove CVE Patch? | no

## Changelog

- Please make sure you have checked our [Contributing guidelines](https://github.com/usabilla/php-docker-template/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: src/php/utils/docker/docker-php-dev-mode [hunk 1]
```diff
@@ -29,7 +29,15 @@ case "$1" in
 			exit 1
 		fi
 
-		pecl install xdebug-stable || pecl install xdebug-2.9.8
+		# If php version >= 8.0 install the right version of xdebug and linux headers
+		php_version=$(php -r 'echo PHP_VERSION;' | cut -c1-3 | tr -d '\n')
+		if [ "$(echo "$php_version >= 8.0" | bc)" = 1 ]; then
+			apk add --update linux-headers
+			pecl install xdebug
```
Comment (ricardofontanelli) [inline src/php/utils/docker/docker-php-dev-mode:36]: Would be better to keep `xdebug-stable` here? I'm not sure if the `xdebug` includes beta versions too.

**Review by ricardofontanelli — APPROVED** [review]

File: src/php/utils/docker/docker-php-dev-mode (hunk 1 above)
Comment (marcogiorgio) [reply src/php/utils/docker/docker-php-dev-mode:36]: ok, let me edit it and see what happens

File: src/php/utils/docker/docker-php-dev-mode [hunk 2]
```diff
@@ -29,7 +29,15 @@ case "$1" in
 			exit 1
 		fi
 
-		pecl install xdebug-stable || pecl install xdebug-2.9.8
+		# If php version >= 8.0 install the right version of xdebug and linux headers
+		php_version=$(php -r 'echo PHP_VERSION;' | cut -c1-3 | tr -d '\n')
+		if [ "$(echo "$php_version >= 8.0" | bc)" = 1 ]; then
+			apk add --update linux-headers
+			pecl install xdebug-stable
```
Comment (carusogabriel) [inline src/php/utils/docker/docker-php-dev-mode:36]: Shall we pin a version < 3.2.0 to avoid the issues we are experiencing?

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 /tmp
 /test/functional/web/tmp
+.idea/*
```
Comment (carusogabriel) [inline .gitignore:3]: small 💅🏼 - you could add this **your** local `.gitignore`, and all the projects using JetBrains' tools won't dirty the projects' `.gitignore`

**Review by carusogabriel — APPROVED** [review]

File: src/php/utils/docker/docker-php-dev-mode [hunk 4]
```diff
@@ -53,4 +61,4 @@ case "$1" in
 		usage
 		exit 1
 		;;
-esac
+esac
```
Comment (WyriHaximus) [inline src/php/utils/docker/docker-php-dev-mode:64]: Small nitpick
```suggestion
esac

```

File: src/php/utils/docker/docker-php-dev-mode (hunk 2 above)
Comment (ricardofontanelli) [reply src/php/utils/docker/docker-php-dev-mode:36]: @carusogabriel the real problem today I think was the `linux-headers` [here ](https://gitlab.usabilla.net/usabilla/dev/account-management/-/jobs/2414206#L2662)

Comment (marcogiorgio) [reply src/php/utils/docker/docker-php-dev-mode:36]: so may I merge the PR?

Comment (WyriHaximus) [reply src/php/utils/docker/docker-php-dev-mode:36]: @ricardofontanelli That would still fail if we don't have those headers installed tho

**Pull request merged**